### PR TITLE
Distributed tracing guidance #231

### PR DESCRIPTION
**Pull request by markweitzel**
Initial pass at telemetry section including open telemetry and reference to azure core.

### CODE REVIEW

File: azure/Guidelines.md [hunk 1]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
-o	Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
+* GET: if the comparison fails, return 304 Not Modified (consider also returning Expires/Cache-Control/Age headers for caching scenarios)
+* PUT/POST/DELETE: if the comparison fails, return 412 Precondition Not Met
+* Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
 
 -	If-Match header - should support multiple values that are Or-ed together per HTTP/1.1.
 
 
+### Distributed Tracing & Telemetry
+Azure Core guidelines specify that you support telemetry data through both the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry.
```
Comment (tg-msft) [inline azure/Guidelines.md]: Most people in the company don't know what "Azure Core" refers to and would probably assume you meant "Core Azure" which is way different.
```suggestion
Azure SDK client guidelines specify that client libraries must send telemetry data through the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry. 
```

File: azure/Guidelines.md [hunk 2]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
+* GET: if the comparison fails, return 304 Not Modified (consider also returning Expires/Cache-Control/Age headers for caching scenarios)
+* PUT/POST/DELETE: if the comparison fails, return 412 Precondition Not Met
+* Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
 
 -	If-Match header - should support multiple values that are Or-ed together per HTTP/1.1.
 
 
+### Distributed Tracing & Telemetry
+Azure Core guidelines specify that you support telemetry data through both the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry. 
+When a client library is created, code to handle telemetry and distributed tracing information will be automatically added. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
```
Comment (tg-msft) [inline azure/Guidelines.md]: Is that true across languages?  The default pipeline does this automatically in languages that have a default pipeline, but I think languages like Java would require you to include this yourself.

Comment (markweitzel) [reply azure/Guidelines.md]: Are there any other languages besides Java where this is not added automatically? I can update this wording a bit to reflect this. Also, and just to confirm we are on the same page, the fact the tooling does not handle this OOTB, does not change/alter the guidance of supporting both of headers.

Comment (tg-msft) [reply azure/Guidelines.md]: We'd have to loop in folks from JS/Python/C++/Go/etc. to confirm as this is a lower level detail.  What if we just changed the wording and leave what clients do/don't do by default up to the language specific client guidelines?

```suggestion
Client libraries are required to send telemetry and distributed tracing information on every  request. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
```

File: azure/Guidelines.md [hunk 3]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
+* Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
 
 -	If-Match header - should support multiple values that are Or-ed together per HTTP/1.1.
 
 
+### Distributed Tracing & Telemetry
+Azure SDK client guidelines specify that client libraries must send telemetry data through the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry. 
+When a client library is created, code to handle telemetry and distributed tracing information will be automatically added. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
 
+:white_check_mark: **DO** follow the Azure Core guidelines for supporting telemetry headers and Open Telemetry.
```
Comment (tg-msft) [inline azure/Guidelines.md]:
```suggestion
:white_check_mark: **DO** follow the Azure SDK client guidelines for supporting telemetry headers and Open Telemetry.
```

File: azure/Guidelines.md [hunk 4]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
 
+:no_entry: **DO NOT** reject a call if you have custom headers you don't understand, and specifically, distributed tracing headers. 
+
+#### Additional References
+* [Azure Core Guidelines](https://azure.github.io/azure-sdk/general_azurecore.html)
```
Comment (tg-msft) [inline azure/Guidelines.md]:
```suggestion
* [Azure SDK client guidelines](https://azure.github.io/azure-sdk/general_azurecore.html)
```

File: azure/Guidelines.md [hunk 5]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
+* [Azure User-Agent header policy](https://azure.github.io/azure-sdk/general_azurecore.html#azurecore-http-telemetry-x-ms-useragent)
```
Comment (tg-msft) [inline azure/Guidelines.md]:
```suggestion
* [Azure SDK User-Agent header policy](https://azure.github.io/azure-sdk/general_azurecore.html#azurecore-http-telemetry-x-ms-useragent)
```

File: azure/Guidelines.md [hunk 6]
```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
...
+* [Azure Distributed tracing policy](https://azure.github.io/azure-sdk/general_azurecore.html#distributed-tracing-policy)
```
Comment (tg-msft) [inline azure/Guidelines.md]:
```suggestion
* [Azure SDK Distributed tracing policy](https://azure.github.io/azure-sdk/general_azurecore.html#distributed-tracing-policy) 
```

File: azure/Guidelines.md (hunk 2 above)
Comment (markweitzel) [reply azure/Guidelines.md]: Yep. I like this. I'll commit this change.

**Comment by markweitzel** [comment]
Thanks @tg-msft - Merging now and closing the PR.

**Pull request merged**